src/bpftrace.cpp

```diff
@@ -1957,21 +1959,45 @@ std::string BPFtrace::resolve_uid(uintptr_t addr) const
   return username;
 }
 
-std::string BPFtrace::resolve_timestamp(uint32_t strftime_id,
-                                        uint64_t nsecs_since_boot)
+std::string BPFtrace::resolve_timestamp(uint32_t mode,
+                                        uint32_t strftime_id,
+                                        uint64_t nsecs)
 {
   static const auto usec_regex = std::regex("%f");
+  TimestampMode ts_mode = static_cast<TimestampMode>(mode);
+  struct timespec zero = {};
+  struct timespec *basetime = &zero;
 
-  if (!boottime_)
+  if (ts_mode == TimestampMode::boot)
   {
-    LOG(ERROR) << "Cannot resolve timestamp due to failed boot time calcuation";
-    return "(?)";
+    if (!boottime_)
+    {
+      LOG(ERROR)
+          << "Cannot resolve timestamp due to failed boot time calcuation";
+      return "(?)";
+    }
+    else
+    {
+      basetime = boottime_.operator->();
```

Comment:
Maybe better
```suggestion
      basetime = &boottime_.value();
```

Reply:
OK, edited. In addition, the `support sw_tai timestamp` patch has also been modified. Because currently nsecs("sw_tai") will return boot time instead of tai time. So put the delta of the timestamp in the codegen stage instead of resolve_timestamp.